src/libcore/iter/mod.rs

```diff
@@ -694,6 +694,18 @@ impl<I> Iterator for StepBy<I> where I: Iterator {
             (f(inner_hint.0), inner_hint.1.map(f))
         }
     }
+
+    #[inline]
+    fn nth(&mut self, mut n: usize) -> Option<Self::Item> {
+        if self.first_take {
+            if n == 0 {
+                self.first_take = false;
+                return self.iter.next()
+            }
+            n -= 1;
+        }
+        self.iter.nth(n * self.step)
```

Comment:
Good change! :heart:

I think that this needs to handle overflow in the multiplication, though, since I could write `(0u64..).step_by(0x10000).nth(0x10000)` on a 32-bit machine, for example.

Also, tests?

Reply:
Tests are a good call, I found 2 off-by-ones while implementing them.

I added overflow behaviour, but it is pretty weighty. While I think that it shouldn't influence normal execution due to `intrinsics::likely` and branch prediction, I do think that it is pretty complicated code, which doesn't really add too much value. Anyway, now that it exists, I don't mind keeping it :)